dlrover/python/master/hyperparams/simple_strategy_generator.py

```diff
@@ -81,68 +62,78 @@ def generate_opt_strategy(
                     gpu_stats, model_config, paral_config.dataloader
                 )
                 optimizer_config = self._generate_optimizer_config()
-                paral_configs[node.name] = ParallelConfig(
+                paral_configs[node.id] = ParallelConfig(
                     data_loader_config, optimizer_config
                 )
-        return paral_configs["simple_node"]
+                node.paral_config = paral_configs[node.id]
+        if paral_configs == {}:
+            logger.info("paral_configs length is 0")
+            return None
+        else:
+            logger.info(f"paral_configs: {paral_configs}")
+            return paral_configs[0]
 
     def _generate_dataloader_config(
         self, gpu_stats, model_config, dataloader_config
     ):
         if gpu_stats == []:
-            return DataLoaderConfig(0, "", 0, 0, 0)
+            return dataloader_config
         # Calculate the minimum remaining memory among GPUs
         min_remain_memory = min(
-            entry["total_memory_gb"] - entry["used_memory_gb"]
-            for entry in gpu_stats
+            entry.total_memory_mb - entry.used_memory_mb for entry in gpu_stats
         )
-
-        # Update dataloader configuration version
-        updated_version = dataloader_config.version + 1
-        # Extract dataloader configuration values
-        batch_size = dataloader_config.batch_size
-
-        # Extract model configuration values
-        block_size = model_config["block_size"]
-        n_layer = model_config["n_layer"]
-        n_heads = model_config["n_heads"]
-        n_embd = model_config["n_embd"]
-
-        # Calculate the memory required for intermediate activation
-        activation_memory_gb = (
-            (
-                34 * batch_size * block_size * n_embd
-                + 5 * batch_size * (block_size**2) * n_heads
+        if min_remain_memory > 2400:
+            # Update dataloader configuration version
+            updated_version = dataloader_config.version + 1
+            # Extract dataloader configuration values
+            batch_size = dataloader_config.batch_size
+
+            # Extract model configuration values
+            block_size = model_config["block_size"]
+            n_layer = model_config["n_layer"]
+            n_heads = model_config["n_heads"]
+            n_embd = model_config["n_embd"]
+
+            # Calculate the memory required for intermediate activation
+            activation_memory_mb = (
+                (
+                    34 * batch_size * block_size * n_embd
+                    + 5 * batch_size * (block_size**2) * n_heads
+                )
+                * n_layer
+                / (1024**2)
             )
-            * n_layer
-            / (1024 * 1024 * 1024)
-        )
-
-        try:
-            updated_batch_size = int(
-                batch_size
-                + batch_size * min_remain_memory / activation_memory_gb
+            try:
```

Comment:
```
if activation_memory_mb > 0:
      updated_batch_size = int(
          batch_size
          + batch_size * min_remain_memory / activation_memory_mb
      )
else:
    updated_batch_size = batch_size
```

Should we keep the batch_size as an even number?

Reply:
I am not sure a even batch_size would be better or not.

Reply:
There are multiple libraries optimized for batch_size in the power of 2. (e.g. 2、4、8、16、32). It‘s reasonable to set an even batch size. Let's make it a `good first issue` to fix.

Reply:
A new issue #706 was created for this problem.